src/sagas/dependency.saga.js

```diff
@@ -186,14 +186,22 @@ export function* handleLoadDependencyInfoFromDiskStart({
 }
 
 // helpers
+function filterYarnInstallMessages(message: string) {
+  // only get [1/4] ... messages and strip warnings (warnings are logged before [1/4])
+  // would be nice if we could show the warnings separately later - for now stripping is OK.
+  const matches = /\[.*/.exec(message);
+  return matches && matches[0];
+}
+
 export function* watchInstallMessages(channel: any): any {
   let output;
   try {
     while (true) {
       output = yield take(channel);
       if (!output.hasOwnProperty('exit')) {
         // Not the final message
-        yield put(setStatusText(output.data));
+        const installMsg = filterYarnInstallMessages(output.data);
```

Comment:
Flow fix because for me the warnings are on `output.error` so for a warning `output.data` is undefined:

```suggestion
        const installMsg = output.data && filterYarnInstallMessages(output.data);
```

Did you get any warnings that are in `output.data`? I wonder if this filter might be unnecessary if all the warnings are getting logged to `output.error` in the end...

<img width="396" alt="Screen Shot 2019-04-02 at 7 01 47 PM" src="https://user-images.githubusercontent.com/17421347/55447732-c8d02a00-5579-11e9-897e-b6389eceb0c7.png">

Reply:
where do you get these warnings? I don't see them. Do you get them in app console or during `yarn flow` run?

Reply:
Just from console.logging the output when I was investigating the Flow error, like this:

```js
// Not the final message
console.log(output);
yield put(setStatusText(output.data));
const installMsg = filterYarnInstallMessages(output.data);
yield put(setStatusText(installMsg));
```

When it hit the fourth output (the warning from my screenshot above), `output.data` was undefined (because the warning is on the error property) so was getting fed into `filterYarnInstallMessages` as undefined, and returning null (hence the Flow error). 

So that's why I'm thinking we may not need the `filterYarnInstallMessages` if warnings are getting spit out as `output.error` anyway (which technically we'd ignore since we're only looking at `output.data`).

Reply:
I think the check for `output.data` is OK to add. I'll add it. The filter is needed because there are also information that are logged to data.

![grafik](https://user-images.githubusercontent.com/3046542/55517775-db2d8f00-5671-11e9-92a6-bb217702ad13.png)

Reply:
Ah gotcha, didn't see any info stuff so missed that 👍